File: s3-inf-ppo/laboratoria/lab07/php/src/Entities/Pedestrian.php
<?php

namespace ParkingLotManager\Entities;

use ParkingLotManager\Interfaces\EntityInterface;

class Pedestrian implements EntityInterface {

	protected $name;

	public function __construct(string $name = "") {
		$this->name = $name;
	}

	public function identify(): string {
		return $this->name ?: "Unknown pedestrian";
	}

	public function canEnter(): bool {
		return true;
	}
	
}

File: s3-inf-ppo/laboratoria/lab07/java/ppo7/src/ParkingLotManager/Entities/Pedestrian.java
package ParkingLotManager.Entities;

import ParkingLotManager.Interfaces.EntityInterface;

public class Pedestrian implements EntityInterface {

    private String name = "";

    public Pedestrian() {}

    public Pedestrian(String name) {
        this.name = name;
    }

    public String identify() {
        return !name.isEmpty() ? name : "Unknown pedestrian";
    }

    public boolean canEnter() {
        return true;
    }

}
